rosa_sts_managed_oidc/main.tf

```diff
@@ -0,0 +1,97 @@
+#
+# Copyright (c) 2023 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Create the VPC if it is wanted
+module "vpc" {
+  create_vpc           = var.create_vpc
+  source               = "../modules/network"
+  aws_region           = var.aws_region
+  vpc_name             = var.vpc_name
+  additional_tags      = var.additional_tags
+  vpc_cidr_block       = var.vpc_cidr_block
+  availability_zones   = var.availability_zones
+  private_subnet_cidrs = var.private_subnet_cidrs
+  public_subnet_cidrs  = var.public_subnet_cidrs
+  single_nat_gateway   = var.single_nat_gateway
+}
+
+# Create the main account roles for ROSA
+module "account_role" {
+  create_account_roles   = var.create_account_roles
+  source                 = "../modules/account_roles"
+  account_role_prefix    = var.account_role_prefix
+  path                   = var.path
+  ocm_environment        = var.ocm_environment
+  rosa_openshift_version = var.rosa_openshift_version
+  account_role_policies  = var.account_role_policies
+  all_versions           = var.all_versions
+  operator_role_policies = var.operator_role_policies
+  additional_tags        = var.additional_tags
+}
+
+# This is unfortunately needed as the cluster will try to build before the account role
+# is created. Hopefully this is fixed in the future
+resource "time_sleep" "wait_10_seconds" {
+  depends_on = [module.account_role]
+
+  create_duration = "10s"
+}
+
+# Create managed OIDC config
+module "oidc_provider" {
+  source               = "../modules/managed_oidc_provider"
```

Comment:
I think it's better to add here the depends_on :
`depends_on = [time_sleep.wait_10_seconds]` 
So the oidc provider will start the creation after 10s delay

Reply:
This isn't needed for the managed oidc provider as it doesn't depend on the account roles as far as I can tell. If you feel there's a risk of a race condition here we can also add a depends_on similar to the example in the unmanaged_oidc_provider example

Reply:
I think it's a good practice... 
Did you test it and it works?

Reply:
Added. Yes, I've done full `terraform init; terraform plan; terraform apply; terraform destroy` with all of these